src/ai/backend/common/pydantic_handlers.py

```diff
@@ -0,0 +1,254 @@
+
+    @property
+    def parsed(self) -> T:
+        if not self._parsed:
+            raise ValueError("Data not yet parsed")
+        return self._parsed
+
+    def from_header(self, headers: CIMultiDictProxy[str]) -> Self:
+        self._parsed = self.model.model_validate(headers)
+        return self
+
+
+class PathParam(Generic[T]):
+    def __init__(self, model: Type[T]) -> None:
+        self.model = model
+        self._parsed: Optional[T] = None
+
+    @property
+    def parsed(self) -> T:
+        if not self._parsed:
+            raise ValueError("Data not yet parsed")
+        return self._parsed
+
+    def from_path(self, match_info: UrlMappingMatchInfo) -> Self:
+        self._parsed = self.model.model_validate(match_info)
+        return self
+
+
+class MiddlewareParam(ABC, BaseModel):
+    @classmethod
+    @abstractmethod
+    def from_request(cls, request: web.Request) -> Self:
+        pass
+
+
+@dataclass
+class BaseResponse:
+    data: BaseModel
+    status_code: int = 200
+
+
+async def _extract_param_value(request: web.Request, input_param_type: Any) -> Optional[Any]:
+    try:
+        # MiddlewareParam Type
+        if get_origin(input_param_type) is None and issubclass(input_param_type, MiddlewareParam):
+            try:
+                return input_param_type.from_request(request)
+            except ValidationError:
+                raise MiddlewareParamParsingFailed(f"Failed while parsing {input_param_type}")
+
+        # If origin type name is BodyParam/QueryParam/HeaderParam/PathParam
+        origin_type = get_origin(input_param_type)
+        pydantic_model = get_args(input_param_type)[0]
+        param_instance = input_param_type(pydantic_model)
+
+        if origin_type is BodyParam:
+            if not request.can_read_body:
+                raise MalformedRequestBody(
+                    f"Malformed body - URL: {request.url}, Method: {request.method}"
+                )
+            try:
+                body = await request.json()
+            except json.decoder.JSONDecodeError:
+                raise MalformedRequestBody(
+                    f"Malformed body - URL: {request.url}, Method: {request.method}"
+                )
+            return param_instance.from_body(body)
+
+        elif origin_type is QueryParam:
+            return param_instance.from_query(request.query)
+
+        elif origin_type is HeaderParam:
+            return param_instance.from_header(request.headers)
+
+        elif origin_type is PathParam:
+            return param_instance.from_path(request.match_info)
+
+        else:
+            raise InvalidAPIParameters(
+                f"Parameter '{input_param_type}' must use one of QueryParam, PathParam, HeaderParam, MiddlewareParam, BodyParam"
+            )
+
+    except ValidationError as e:
+        raise InvalidAPIParameters(str(e))
+
+
+class _HandlerParameters:
+    def __init__(self) -> None:
+        self.params: dict[str, Any] = {}
```

Comment:
Please declare all the fields used in the class.

Reply:
If the value should not be accessed outside the class, please set it as private.